Include/internal/pycore_c_array.h

```diff
@@ -0,0 +1,38 @@
+#ifndef Py_INTERNAL_C_ARRAY_H
+#define Py_INTERNAL_C_ARRAY_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef Py_BUILD_CORE
+#  error "this header requires Py_BUILD_CORE define"
+#endif
+
+
+/* Utility for a number of growing arrays
+ *
+ * idx: index we need to write to
```

Comment:
Would it make sense to have a struct containing the pointers and sizes, like `struct _Py_hashtable_t` does for hashtables.

I have to admit, I'm a bit surprised we don't have this elsewhere.

Reply:
Yes I thought of doing this. Let me try.